applications/nrf_desktop/src/modules/ble_conn_params.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/conn.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/gatt_dm.h>
+
+#ifdef CONFIG_BT_LL_NRFXLIB
+#include "ble_controller_hci_vs.h"
+#endif /* CONFIG_BT_LL_NRFXLIB */
+
+#define MODULE ble_conn_params
+#include "module_state_event.h"
+#include "ble_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BLE_CONN_PARAMS_LOG_LEVEL);
+
+#define CONN_INTERVAL_LLPM_US		1000   /* 1 ms */
+#define CONN_INTERVAL_LLPM_REG		0x0d01 /* 1 ms */
+#define CONN_INTERVAL_BLE_REG		0x0006 /* 7.5 ms */
+#define CONN_LATENCY_LOW		0
+#define CONN_LATENCY_DEFAULT		99
+#define CONN_SUPERVISION_TIMEOUT	400
+
+#define CONN_PARAMS_ERROR_TIMEOUT	K_MSEC(5)
+
+struct connected_peer {
+	struct bt_conn *conn;
+	bool llpm_support;
+	u16_t requested_latency;
+};
+
+static struct connected_peer peers[CONFIG_BT_MAX_CONN];
+static struct k_delayed_work conn_params_update;
+
+
+static struct connected_peer *find_connected_peer(struct bt_conn *conn)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(peers); i++) {
+		if (peers[i].conn == conn) {
+			return &peers[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int set_conn_params(struct bt_conn *conn, u16_t conn_latency,
+			   bool peer_llpm_support)
+{
+	int err;
+
+#ifdef CONFIG_BT_LL_NRFXLIB
+	if (peer_llpm_support) {
+		struct net_buf *buf;
+
+		hci_vs_cmd_conn_update_t *cmd_conn_update;
+
+		buf = bt_hci_cmd_create(HCI_VS_OPCODE_CMD_CONN_UPDATE,
+					sizeof(*cmd_conn_update));
+		if (!buf) {
+			LOG_ERR("Could not allocate command buffer");
+			return -ENOBUFS;
+		}
+
+		u16_t conn_handle;
+
+		err = bt_hci_get_conn_handle(conn, &conn_handle);
+		if (err) {
+			LOG_ERR("Failed obtaining conn_handle (err:%d)", err);
```

Comment:
`buf` is not released on error

Reply:
`net_buf_unref` should be called on it most probably

Reply:
or `bt_hci_get_conn_handle` should be called before the buffer is obtained

Reply:
You are right. I reordered the calls.
It was leftover from previous implementation.